hugegraph-llm/src/hugegraph_llm/demo/rag_demo/log_block.py

```diff
@@ -0,0 +1,162 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import gradio as gr
+from hugegraph_llm.config import settings
+from hugegraph_llm.utils.log import log
+import os
+import asyncio
+import requests
+from fastapi import Request
+# Generator to simulate the tail -f behavior
+async def log_stream(log_path: str):
+    """
+    Stream the content of a log file like `tail -f`.
+    This function is asynchronous.
+    """
+    try:
+        with open(log_path, 'r') as file:
+            # Move the cursor to the end of the file
+            # file.seek(0, os.SEEK_END)
+            while True:
+                line = file.readline()
+                if line:
+                    yield line
+                else:
+                    await asyncio.sleep(0.1)  # Non-blocking sleep
+    except FileNotFoundError:
+        raise Exception(f"Log file not found: {log_path}")
+    except Exception as e:
+        raise Exception(f"An error occurred while reading the log: {str(e)}")
+            
+# Functions to read each log file
+def read_llm_server_log():
+    try:
+        with open("logs/llm-server.log", "r") as f:
+            return f.read()
+    except FileNotFoundError:
+        return "LLM Server log file not found."
+
+def read_output_log():
+    try:
+        with open("logs/output.log", "r") as f:
+            return f.read()
+    except FileNotFoundError:
+        return "Output log file not found."
+
+# Functions to clear each log file
+def clear_llm_server_log():
+    with open("logs/llm-server.log", "w") as f:
+        f.truncate(0)  # Clear contents of the file
+    return "LLM Server log cleared."
+
+def clear_output_log():
+    with open("logs/output.log", "w") as f:
+        f.truncate(0)  # Clear contents of the file
+    return "Output log cleared."
+
+# Function to validate password and control access to logs
+def check_password(password, request:Request):
+    client_ip = request.client.host if request else "Unknown IP"
+    if password == settings.log_auth_key:
+        # Return logs and update visibility
+        llm_log, output_log = read_llm_server_log(), read_output_log()
+        visible_update = gr.update(visible=True)
+        hidden_update = gr.update(visible=False)
+
+        # Log the successful access with the IP address
+        log.info(f"Logs accessed successfully from IP: {client_ip}")
+
+        return llm_log, output_log, visible_update, visible_update, visible_update, hidden_update
+    else:
+        # Log the failed attempt with IP address
+        log.error(f"Incorrect password attempt from IP: {client_ip}")
+        return "", "", gr.update(visible=False), gr.update(visible=False), gr.update(visible=False), gr.update(value="Incorrect password. Access denied.", visible=True)
```

Comment:
Note: max_chars for one line is **120** now (py code)

so as other modification